File: core-java-8/src/main/java/com/baeldung/datetime/UseLocalTime.java
package com.baeldung.datetime;

import java.time.LocalTime;
import java.time.temporal.ChronoUnit;

public class UseLocalTime {
    
    public LocalTime getLocalTimeUsingFactoryOfMethod(int hour, int min, int seconds){        
        LocalTime localTime = LocalTime.of(hour, min, seconds);
        return localTime;
    }
    
    public LocalTime getLocalTimeUsingParseMethod(String timeRepresentation){        
        LocalTime localTime = LocalTime.parse(timeRepresentation);
        return localTime;
    }
    
    public LocalTime getLocalTimeFromClock(){        
        LocalTime localTime = LocalTime.now();
        return localTime;
    }
    
    public LocalTime addAnHour(LocalTime localTime){
        LocalTime newTime = localTime.plus(1,ChronoUnit.HOURS);
        return newTime;
    }
    
    public int getHourFromLocalTime(LocalTime localTime){        
        return localTime.getHour();
    }
    
    public LocalTime getLocalTimeWithMinuteSetToValue(LocalTime localTime, int minute){        
        return localTime.withMinute(minute);
    }
}

File: core-kotlin-modules/core-kotlin-2/src/main/kotlin/com/baeldung/dates/datetime/UseLocalTime.kt
package com.baeldung.dates.datetime

import java.time.LocalTime
import java.time.temporal.ChronoUnit

class UseLocalTime {

    fun getLocalTimeUsingFactoryOfMethod(hour: Int, min: Int, seconds: Int): LocalTime {
        return LocalTime.of(hour, min, seconds)
    }

    fun getLocalTimeUsingParseMethod(timeRepresentation: String): LocalTime {
        return LocalTime.parse(timeRepresentation)
    }

    fun getLocalTimeFromClock(): LocalTime {
        return LocalTime.now()
    }

    fun addAnHour(localTime: LocalTime): LocalTime {
        return localTime.plus(1, ChronoUnit.HOURS)
    }

    fun getHourFromLocalTime(localTime: LocalTime): Int {
        return localTime.hour
    }

    fun getLocalTimeWithMinuteSetToValue(localTime: LocalTime, minute: Int): LocalTime {
        return localTime.withMinute(minute)
    }
}